### ISSUE TITLE
jest/vitest rules: allow configuring additional test-file patterns (e.g. `.steps.ts`/`.helper.ts` BDD helpers)

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25636]
## Summary

oxlint runs jest/vitest rules only on files its built-in heuristic classifies as test files: a `__tests__/` path component, a `*.test.*`/`*.spec.*` filename, or an import of `vitest`/`@jest/globals`-family modules. Codebases with BDD-style helper files (`*.steps.ts`, `*.helper.ts`) that contain `test()`/`expect()` calls silently lose all jest-rule coverage on those files — the rules never run, and nothing warns that they were skipped. ESLint has no such gating: eslint-plugin-jest rules run wherever the config enables them.

## oxlint version

1.78.0 (also verified on 1.69.0–1.72.0; the classification lives in `crates/oxc_linter/src/frameworks.rs` and has no config surface).

## Reproduction

Identical content in three files:

```ts
// a.test.ts, a.steps.ts, a.helper.ts — identical:
test('has no assertions', () => {
	const x = 1 + 1
})
```

`.oxlintrc.json`:

```json
{"plugins": ["jest"], "categories": {"correctness": "off"}, "rules": {"jest/expect-expect": "error"}}
```

```
$ oxlint -c .oxlintrc.json --disable-nested-config a.test.ts a.steps.ts a.helper.ts
a.test.ts:1:1: error jest(expect-expect): Test has no assertions
```

Only `a.test.ts` is diagnosed; `a.steps.ts` and `a.helper.ts` produce nothing.

## Ask

A config knob to widen the classification — for example:

```json
{"settings": {"jest": {"additionalTestPatterns": ["**/*.steps.ts", "**/*.helper.ts"]}}}
```

mirroring how `settings.jest.version` already exists, or an equivalent per-override mechanism.

## Real-world impact

A large monorepo migrating from ESLint had jest-rule coverage (via eslint-plugin-jest) on ~25 e2e helper files (`.steps.ts`/`.helper.ts` containing `expect()` assertions used by BDD-style specs) and lost it in the oxlint migration — 4 rules affected (`expect-expect`, `no-standalone-expect`, `valid-expect`, `no-restricted-matchers`), with no way to restore coverage short of renaming the files into the recognized patterns.
